### PR TITLE
doc: nrf_modem: updates to delta DFU

### PR DESCRIPTION
**Pull request by lemrey**
Update the delta DFU documentation with flow charts for both the update and rollback operations. Other minor updates.

### CODE REVIEW

**Review by jarlamsa — APPROVED** [review]
LGTM

**Comment by Avalei** [comment]
It looks like the figures may have been made using an older template. I went ahead and committed an update using our current figure template so that they would be size and style-compliant.

**Review by Avalei — APPROVED** [review]

File: nrf_modem/doc/delta_dfu.rst [hunk 1]
```diff
@@ -172,8 +176,10 @@ The application schedules the execution of a firmware delta update by calling :c
 Upon success, the operation is scheduled and the modem updates its firmware the next time the application boots the modem by calling :c:func:`nrf_modem_init`.
 
 .. note::
-   The return value of :c:func:`nrf_modem_delta_dfu_update` only indicates that the modem has scheduled the update and does not represent the result of the actual update operation.
-   The result is returned by :c:func:`nrf_modem_init`.
+   The return value of :c:func:`nrf_modem_delta_dfu_update` only indicates whether the modem has scheduled the update and does not represent the result of the actual update operation.
+
+.. figure:: images/delta_dfu_update.svg
+   :alt: Modem delta DFU update flowchart
```
Comment (peknis) [inline nrf_modem/doc/delta_dfu.rst:182]:
```suggestion
.. figure:: images/delta_dfu_update.svg
   :alt: Modem delta DFU update flowchart
   
   Modem delta DFU update flowchart
```
This adds the caption.

File: nrf_modem/doc/delta_dfu.rst [hunk 2]
```diff
@@ -199,25 +205,46 @@ The application can reinitialize the Modem library by calling :c:func:`nrf_modem
...
 
-.. note::
-   The rollback functionality is only available as long as the application does not erase the contents of the scratch area by calling :c:func:`nrf_modem_delta_dfu_erase`.
-   Once the application has called :c:func:`nrf_modem_delta_dfu_erase` it can no longer roll back to a previous modem firmware version using the delta DFU API.
+The figure below shows the flow chart for the rollback operation.
 
-.. note::
-   The modem must be shut down by calling :c:func:`nrf_modem_shutdown` prior to calling :c:func:`nrf_modem_init`.
+.. figure:: images/delta_dfu_rollback.svg
+   :alt: Modem delta DFU rollback flowchart
```
Comment (peknis) [inline nrf_modem/doc/delta_dfu.rst:226]:
```suggestion
.. figure:: images/delta_dfu_rollback.svg
   :alt: Modem delta DFU rollback flowchart
   
   Modem delta DFU rollback flowchart
```
Adding the caption.

**Comment by lemrey** [comment]
Compliance issues will disappear after we squash the commits.

File: nrf_modem/doc/delta_dfu.rst [hunk 3]
```diff
@@ -199,25 +205,46 @@ The application can reinitialize the Modem library by calling :c:func:`nrf_modem
...
+* ``MODEM_DFU_RESULT_INTERNAL_ERROR`` - The modem encountered an internal error while executing the rollback, and it will not boot to prevent executing unintended operations. For subsequent programming, the modem can only be programmed through the :ref:`nrf_modem_full_dfu_api`.
+* ``MODEM_DFU_RESULT_HARDWARE_ERROR`` - The modem encountered a hardware error while executing the rollback, and it will not boot to prevent executing unintended operations. For subsequent programming, the modem can only be programmed through the :ref:`nrf_modem_full_dfu_api`.
+
+Reinitializing the modem to run the firmware
+============================================
+
+Regardless of the value returned by :c:func:`nrf_modem_init`, the application must reinitialize the modem by reinitializing the Modem library in order to run the modem firmware.
+The application can reinitialize the Modem library by calling :c:func:`nrf_modem_shutdown` followed by :c:func:`nrf_modem_init`.
+If the rollback is successful, the modem runs the previous modem firmware.
+The application can verify that by reading the modem firmware UUID or reading the ``AT+CGMR`` command response.
```
Comment (peknis) [inline nrf_modem/doc/delta_dfu.rst:250]:
```suggestion
The application can verify that by reading the modem firmware UUID or the ``AT+CGMR`` command response.
```

File: nrf_modem/doc/delta_dfu.rst [hunk 4]
```diff
@@ -199,25 +205,46 @@ The application can reinitialize the Modem library by calling :c:func:`nrf_modem
...
+* ``MODEM_DFU_RESULT_AUTH_ERROR`` - The rollback did not take place. The modem will run the same firmware the next time it boots.
+* ``MODEM_DFU_RESULT_UUID_ERROR`` - The rollback did not take place. The modem will run the same firmware the next time it boots.
+* ``MODEM_DFU_RESULT_INTERNAL_ERROR`` - The modem encountered an internal error while executing the rollback, and it will not boot to prevent executing unintended operations. For subsequent programming, the modem can only be programmed through the :ref:`nrf_modem_full_dfu_api`.
+* ``MODEM_DFU_RESULT_HARDWARE_ERROR`` - The modem encountered a hardware error while executing the rollback, and it will not boot to prevent executing unintended operations. For subsequent programming, the modem can only be programmed through the :ref:`nrf_modem_full_dfu_api`.
+
+Reinitializing the modem to run the firmware
+============================================
+
+Regardless of the value returned by :c:func:`nrf_modem_init`, the application must reinitialize the modem by reinitializing the Modem library in order to run the modem firmware.
```
Comment (peknis) [inline nrf_modem/doc/delta_dfu.rst:247]:
```suggestion
Regardless of the value returned by :c:func:`nrf_modem_init`, the application must reinitialize the modem by reinitializing the :ref:`nrf_modem` to run the modem firmware.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by lemrey** [comment]
@peknis I have committed your suggestions

**Review by peknis — APPROVED** [review]

**Comment by lemrey** [comment]
I picked this commit in #634

**Pull request closed without merging**